app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,29 @@
+class Api::V1::ReservationsController < ApplicationController
+  def index
+    user = current_user
+    @reservations = Reservation.includes(:tour).where(user_id: user.id)
+    render json: @reservations
+  end
+
+  def show
+    @reservation = Reservation.find_by(id: params[:id])
+    render json: @reservation
+  end
+
+  def create
+    @user = current_user
+    @reservation = Reservation.create(reservation_params)
+    @reservation.user_id = @user.id
+    if @reservation.save
+      render json: @reservation, status: :created, notice: 'Reservation created successfully'
```

Comment:
I suggest you use `new` as opposed to `create` on line **15**, because it is not a good practice to use create and save together, as create already saved the record to the database. 
Check the example snippet below:

```ruby
def create
    @reservation = current_user.reservations.new(reservation_params)
    if @reservation.save
```